pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PinotURIUploadIntegrationTest.java

```diff
@@ -186,6 +187,34 @@ private void verifyNRows(int currentNrows, int finalNrows)
     Assert.fail("Failed to get from " + currentNrows + " to " + finalNrows);
   }
 
+  @Test(dataProvider = "configProvider")
+  public void testSegmentValidator(String tableName, SegmentVersion version) {
+    completeTableConfiguration();
+    String serverInstanceId = "Server_localhost_" + CommonConstants.Helix.DEFAULT_SERVER_NETTY_PORT;
+
+    // Disable server instance.
+    _helixAdmin.enableInstance(getHelixClusterName(), serverInstanceId, false);
+
+    final String segment6 = "segmentToBeRefreshed_6";
+    final int nRows1 = 69;
+    try {
+      generateAndUploadRandomSegment(segment6, nRows1);
+      Assert.fail("Uploading segments should fail.");
+    } catch (Exception e) {
+      Assert.assertNotNull(e);
+      Assert.assertTrue(e.getMessage().contains("No assigned Instances for Segment"));
+    }
+
+    // Re-enable the server instance.
+    _helixAdmin.enableInstance(getHelixClusterName(), serverInstanceId, true);
+
+    try {
+      generateAndUploadRandomSegment(segment6, nRows1);
```

Comment:
Can we split generate and upload so we can reuse the same segment here?

Reply:
Done.